vpn/src/main/java/com/duckduckgo/mobile/android/vpn/prefs/VpnSharedPreferencesProvider.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.mobile.android.vpn.prefs
+
+import android.content.Context
+import android.content.Context.MODE_PRIVATE
+import android.content.SharedPreferences
+import com.duckduckgo.di.scopes.AppScope
+import com.frybits.harmony.getHarmonySharedPreferences
+import com.squareup.anvil.annotations.ContributesBinding
+import javax.inject.Inject
+
+interface VpnSharedPreferencesProvider {
+    fun getSharedPreferences(name: String, multiprocess: Boolean = false): SharedPreferences
+}
```

Comment:
This is only in the `:vpn` module for now. If we like this pattern we can move it to eg. `:common` module and use it across the app (which I think we should). But it's kind of out of scope of this PR I think

Reply:
I personally like the idea.